src/YDialogHelpers.cc

```diff
@@ -0,0 +1,233 @@
+using std::vector;
+
+
+void
+YDialog::showText( const string & text, bool useRichText )
+{
+
+    // set help text dialog size to 80% of topmost dialog, respectively 45x15 (default)
+
+    unsigned int dialogWidth  = 45;
+    unsigned int dialogHeight = 15;
+
+    if ( ! _dialogStack.empty() )
+    {
+	YDialog * dialog = _dialogStack.top();
+        dialogWidth  = (unsigned int) ( (float) dialog->preferredWidth()  * 0.8 );
+        dialogHeight = (unsigned int) ( (float) dialog->preferredHeight() * 0.8 );
+    }
+
+    // limit dialog to a reasonable size
+    if ( dialogWidth > 80 || dialogHeight > 25 )
+    {
+	dialogWidth = 80;
+	dialogHeight = 25;
+    }
+
+    try
+    {
+	YDialog     * dialog    = YUI::widgetFactory()->createPopupDialog();
+	YAlignment  * minSize   = YUI::widgetFactory()->createMinSize( dialog, dialogWidth, dialogHeight );
+	YLayoutBox  * vbox      = YUI::widgetFactory()->createVBox( minSize );
+	YUI::widgetFactory()->createRichText( vbox, text, ! useRichText );
+	YButtonBox  * buttonBox = YUI::widgetFactory()->createButtonBox( vbox );
+	YPushButton * okButton  = YUI::widgetFactory()->createPushButton( buttonBox, "&OK" );
+	okButton->setRole( YOKButton );
+	okButton->setDefaultButton();
+
+	dialog->waitForEvent();
+	dialog->destroy();
+    }
+    catch ( const YUIException & exception )
+    {
+	// Don't let the application die just because help couldn't be displayed.
+
+	YUI_CAUGHT( exception );
+    }
+}
+
+
+bool
+YDialog::showHelpText( YWidget * widget )
+{
+    string helpText;
+
+    while ( widget )
+    {
+	if ( ! widget->helpText().empty() )
+	{
+	    yuiDebug() << "Found help text for " << widget << endl;
+	    helpText = widget->helpText();
+	}
+
+	widget = widget->parent();
+    }
+
+    if ( ! helpText.empty() )
+    {
+	yuiMilestone() << "Showing help text" << endl;
+	showText( helpText, true );
+
+	yuiMilestone() << "Help dialog closed" << endl;
+    }
+    else // No help text
+    {
+	yuiWarning() << "No help text" << endl;
+    }
+
+    return ! helpText.empty();
+}
+
+
+bool
+YDialog::showRelNotesText()
+{
+    yuiMilestone() <<"Showing Release Notes" << endl;
+
+    // set help text dialog size to 80% of topmost dialog, respectively 45x15 (default)
+
+    unsigned int dialogWidth  = 45;
+    unsigned int dialogHeight = 15;
+
+    if ( ! _dialogStack.empty() )
+    {
+        YDialog * dialog = _dialogStack.top();
+        dialogWidth  = (unsigned int) ( (float) dialog->preferredWidth()  * 0.8 );
+        dialogHeight = (unsigned int) ( (float) dialog->preferredHeight() * 0.8 );
+    }
+
+    // limit dialog to a reasonable size
+    if ( dialogWidth > 80 || dialogHeight > 25 )
+    {
+        dialogWidth = 80;
+        dialogHeight = 25;
+    }
+
+    try
+    {
+	map<string,string> relnotes = YUI::application()->releaseNotes();
+
+	if ( relnotes.size() == 0)
+	{
+	    return false;
+	}
+
+	vector<string> keys;
+	for ( map<string,string>::iterator it = relnotes.begin(); it != relnotes.end(); ++it )
```

Comment:
This could be a ```const_iterator``` (already in the original location).

Reply:
That code was just moved verbatim to a new file.

Reply:
Fixed.